frame/benchmarking/src/lib.rs

```diff
@@ -408,8 +474,40 @@ macro_rules! benchmarks_iter {
 			$( $rest )*
 		);
 	};
-	// iteration-exit arm
+	// iteration-exit arm which generates a #[test] function for each case.
+	(
+		{ $bench_module:ident, $new_test_ext:expr, $test:path $(, $( $args:tt )* )? }
+		{ $( $instance:ident: $instance_bound:tt )? }
+		{ $( $where_clause:tt )* }
+		( $( $names:tt )* )
+		( $( $names_extra:tt )* )
+		( $( $names_skip_meta:tt )* )
+	) => {
+		$crate::selected_benchmark!(
+			{ $( $where_clause)* }
+			{ $( $instance: $instance_bound )? }
+			$( $names )*
+		);
+		$crate::impl_benchmark!(
+			{ $( $where_clause )* }
+			{ $( $instance: $instance_bound )? }
+			( $( $names )* )
+			( $( $names_extra ),* )
+			( $( $names_skip_meta ),* )
+		);
+		$crate::impl_test_function!(
+			( $( $names )* )
+			( $( $names_extra )* )
+			( $( $names_skip_meta )* )
+			$bench_module,
+			$new_test_ext,
+			$test
+			$(, $( $args )* )?
+		);
+	};
+	// iteration-exit arm which generates one #[test] function for all cases.
```

Comment:
comment is a bit misleading, in this branch no test are generated. and in both branch the `test_bench_by_name` is implemented (inside impl_benchmark)
```suggestion
	// iteration-exit arm which doesn't generates a #[test] function for all cases.
```

Reply:
Ok I see what you mean, it can also be understood that way. Fixed it (minus the `s` at the `generates`).